cmd/kubeadm/app/master/discovery.go

```diff
@@ -116,7 +116,8 @@ func newKubeDiscovery(cfg *kubeadmapi.MasterConfiguration, caCert *x509.Certific
 	}
 
 	SetMasterTaintTolerations(&kd.Deployment.Spec.Template.ObjectMeta)
-	SetNodeAffinity(&kd.Deployment.Spec.Template.ObjectMeta, MasterNodeAffinity(), NativeArchitectureNodeAffinity())
+	// TODO: This does not make sense, deployments don't have "affinity"
```

Comment:
Yeah but it needs to be on the podspec I guess is what I was trying to state.  b4 they were adding the annotation to the deployment object itself.